### PR TITLE
Refactoring greenpithumb.py

### PR DESCRIPTION
**Pull request by mtlynch**
This file had gotten a bit unwieldy with several functions taking a million
args and doing unrelated actions. Refactors to try to break it into more
logically cohesive chunks.

### CODE REVIEW

File: greenpithumb/greenpithumb.py [hunk 1]
```diff
@@ -29,35 +29,135 @@
...
+    return camera_manager.CameraManager(image_path,
+                                        clock.Clock(), camera, light_sensor)
+
+
+def make_pump_manager(moisture_threshold, sleep_windows, raspberry_pi_io,
+                      wiring_config, pump_amount, pump_interval):
+    """Creates a pump manager instance.
+
+    Args:
+        moisture_threshold: The minimum water level before the pump turns on.
```
Comment (JeetShetty) [inline greenpithumb/greenpithumb.py]: The way this is worded is a little confusing. Maybe something like 'The minimum moisture level below which the pump turns on.'

File: greenpithumb/greenpithumb.py [hunk 2]
```diff
@@ -29,35 +29,135 @@
...
+                        temperature_sensor, humidity_sensor,
+                        soil_moisture_sensor, light_sensor, camera_manager,
+                        pump_manager):
+    """Creates a poller for each GreenPiThumb sensor.
+
+    Args:
+        poll_interval: The frequency at which to poll non-camera sensors.
+        photo_interval: The frequency at which to capture photos.
+        record_queue: Queue on which to put sensor reading records.
+        temperature_sensor: Sensor for recording temperature.
```
Comment (JeetShetty) [inline greenpithumb/greenpithumb.py]: (optional) Maybe more accurate to say 'Sensor for capturing temperature'? Ditto for the other sensors.

File: greenpithumb/greenpithumb.py [hunk 3]
```diff
@@ -29,35 +29,135 @@
...
+
+def read_wiring_config(config_filename):
+    """Parses wiring config from a file."""
+    logger.info('reading wiring config at "%s"', config_filename)
+    with open(config_filename) as config_file:
+        return wiring_config_parser.parse(config_file.read())
+
+
+def make_adc(wiring_config):
+    """Creates ADC instance based on the given wiring_config."""
```
Comment (JeetShetty) [inline greenpithumb/greenpithumb.py]: (optional) Noticed you have an Args section in ```make_dht11_sensors``` but not here.

File: greenpithumb/greenpithumb.py [hunk 4]
```diff
@@ -67,33 +167,22 @@ def make_sensor_pollers(poll_interval, wiring_config, moisture_threshold,
...
-            pump_manager), poller_factory.create_light_poller(
-                light_sensor.LightSensor(
-                    adc, wiring_config.adc_channels.light_sensor)),
-        camera_poller_factory.create_camera_poller(
-            camera_manager.CameraManager(
-                image_path, utc_clock, camera,
-                light_sensor.LightSensor(
-                    adc, wiring_config.adc_channels.light_sensor)))
+            soil_moisture_sensor,
+            pump_manager), poller_factory.create_light_poller(light_sensor),
```
Comment (JeetShetty) [inline greenpithumb/greenpithumb.py]: (optional) Weird formatting here.

File: greenpithumb/greenpithumb.py (hunk 1 above)
Comment (mtlynch) [inline greenpithumb/greenpithumb.py]: Ah, good point, fixed.

File: greenpithumb/greenpithumb.py (hunk 2 above)
Comment (mtlynch) [inline greenpithumb/greenpithumb.py]: Oh, true. I feel like "capturing" also kind of implies too much. Changed to "measuring."

File: greenpithumb/greenpithumb.py (hunk 3 above)
Comment (mtlynch) [inline greenpithumb/greenpithumb.py]: Fixed

File: greenpithumb/greenpithumb.py (hunk 4 above)
Comment (mtlynch) [inline greenpithumb/greenpithumb.py]: Oh yeah, that is a little funky. Added a YAPF override to fix it.

**Pull request merged**